surveys/serializers.py

```diff
@@ -6,6 +7,16 @@
 from . import models
 
 
+def update_nested(d, u):
+    for k, v in u.items():
+        if isinstance(v, collections.Mapping):
+            r = update_nested(d.get(k, {}), v)
+            d[k] = r
+        else:
+            d[k] = u[k]
+    return d
```

Comment:
Actually, this could use a docstring and some more descriptive variable names, that'd help.